--- ../src-base/minecraft/net/minecraftforge/fml/common/network/IGuiHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/IGuiHandler.java
@@ -24,17 +24,16 @@
 
 import javax.annotation.Nullable;
 
-public interface IGuiHandler
-{
+public interface IGuiHandler {
     /**
      * Returns a Server side Container to be displayed to the user.
      *
-     * @param ID The Gui ID Number
+     * @param ID     The Gui ID Number
      * @param player The player viewing the Gui
-     * @param world The current world
-     * @param x X Position
-     * @param y Y Position
-     * @param z Z Position
+     * @param world  The current world
+     * @param x      X Position
+     * @param y      Y Position
+     * @param z      Z Position
      * @return A GuiScreen/Container to be displayed to the user, null if none.
      */
     @Nullable
@@ -45,12 +44,12 @@
      * needs to return a instance of GuiScreen On the server side, this needs to
      * return a instance of Container
      *
-     * @param ID The Gui ID Number
+     * @param ID     The Gui ID Number
      * @param player The player viewing the Gui
-     * @param world The current world
-     * @param x X Position
-     * @param y Y Position
-     * @param z Z Position
+     * @param world  The current world
+     * @param x      X Position
+     * @param y      Y Position
+     * @param z      Z Position
      * @return A GuiScreen/Container to be displayed to the user, null if none.
      */
     @Nullable
