--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/IFMLCallHook.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/IFMLCallHook.java
@@ -28,14 +28,13 @@
  * safe environment for further loading of FML.
  *
  * @author cpw
- *
  */
-public interface IFMLCallHook extends Callable<Void>
-{
+public interface IFMLCallHook extends Callable<Void> {
     /**
      * Injected with data from the FML environment:
      * "classLoader" : The FML Class Loader
+     *
      * @param data
      */
-    void injectData(Map<String,Object> data);
+    void injectData(Map<String, Object> data);
 }
