--- ../src-base/minecraft/net/minecraftforge/fml/relauncher/Side.java
+++ ../src-work/minecraft/net/minecraftforge/fml/relauncher/Side.java
@@ -35,16 +35,14 @@
     /**
      * @return If this is the server environment
      */
-    public boolean isServer()
-    {
+    public boolean isServer() {
         return !isClient();
     }
 
     /**
      * @return if this is the Client environment
      */
-    public boolean isClient()
-    {
+    public boolean isClient() {
         return this == CLIENT;
     }
 }
