--- ../src-base/minecraft/net/minecraft/util/text/translation/LanguageMap.java
+++ ../src-work/minecraft/net/minecraft/util/text/translation/LanguageMap.java
@@ -69,61 +69,49 @@
                 }
             }
 
-        }
-        catch (IOException var7)
-        {
+        } catch (IOException var7) {
             ;
+        } catch (Exception ex) {
         }
-        catch (Exception ex) {}
         return table;
     }
 
-    static LanguageMap getInstance()
-    {
+    static LanguageMap getInstance() {
         return instance;
     }
 
     @SideOnly(Side.CLIENT)
 
-    public static synchronized void replaceWith(Map<String, String> p_135063_0_)
-    {
+    public static synchronized void replaceWith(Map<String, String> p_135063_0_) {
         instance.languageList.clear();
         instance.languageList.putAll(p_135063_0_);
         instance.lastUpdateTimeInMilliseconds = System.currentTimeMillis();
     }
 
-    public synchronized String translateKey(String key)
-    {
+    public synchronized String translateKey(String key) {
         return this.tryTranslateKey(key);
     }
 
-    public synchronized String translateKeyFormat(String key, Object... format)
-    {
+    public synchronized String translateKeyFormat(String key, Object... format) {
         String s = this.tryTranslateKey(key);
 
-        try
-        {
+        try {
             return String.format(s, format);
-        }
-        catch (IllegalFormatException var5)
-        {
+        } catch (IllegalFormatException var5) {
             return "Format error: " + s;
         }
     }
 
-    private String tryTranslateKey(String key)
-    {
+    private String tryTranslateKey(String key) {
         String s = this.languageList.get(key);
         return s == null ? key : s;
     }
 
-    public synchronized boolean isKeyTranslated(String key)
-    {
+    public synchronized boolean isKeyTranslated(String key) {
         return this.languageList.containsKey(key);
     }
 
-    public long getLastUpdateTimeInMilliseconds()
-    {
+    public long getLastUpdateTimeInMilliseconds() {
         return this.lastUpdateTimeInMilliseconds;
     }
 }
