--- ../src-base/minecraft/net/minecraftforge/client/event/PlayerSPPushOutOfBlocksEvent.java
+++ ../src-work/minecraft/net/minecraftforge/client/event/PlayerSPPushOutOfBlocksEvent.java
@@ -28,20 +28,23 @@
 
 /**
  * This event is called before the pushOutOfBlocks calls in EntityPlayerSP.
- *
+ * <p>
  * Cancelling the event will prevent pushOutOfBlocks from being called.
  */
 @Cancelable
-public class PlayerSPPushOutOfBlocksEvent extends PlayerEvent
-{
+public class PlayerSPPushOutOfBlocksEvent extends PlayerEvent {
     private AxisAlignedBB entityBoundingBox;
 
-    public PlayerSPPushOutOfBlocksEvent(EntityPlayer player, AxisAlignedBB entityBoundingBox)
-    {
+    public PlayerSPPushOutOfBlocksEvent(EntityPlayer player, AxisAlignedBB entityBoundingBox) {
         super(player);
         this.entityBoundingBox = entityBoundingBox;
     }
 
-    public AxisAlignedBB getEntityBoundingBox() { return entityBoundingBox; }
-    public void setEntityBoundingBox(@Nonnull AxisAlignedBB entityBoundingBox) { this.entityBoundingBox = entityBoundingBox; }
+    public AxisAlignedBB getEntityBoundingBox() {
+        return entityBoundingBox;
+    }
+
+    public void setEntityBoundingBox(@Nonnull AxisAlignedBB entityBoundingBox) {
+        this.entityBoundingBox = entityBoundingBox;
+    }
 }
