--- ../src-base/minecraft/net/minecraftforge/fml/common/event/FMLPostInitializationEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/event/FMLPostInitializationEvent.java
@@ -19,60 +19,51 @@
 
 package net.minecraftforge.fml.common.event;
 
-import java.util.Arrays;
-import java.util.Objects;
-import java.util.Optional;
-
 import net.minecraftforge.fml.common.FMLLog;
 import net.minecraftforge.fml.common.Loader;
 import net.minecraftforge.fml.common.LoaderState.ModState;
 
 import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
 
 /**
  * Called after {@link FMLInitializationEvent} has been dispatched on every mod. This is the third and last
  * commonly called event during mod initialization.
- *
+ * <p>
  * Recommended activities: interact with other mods to establish cross-mod behaviours.
  *
- * @see net.minecraftforge.fml.common.Mod.EventHandler for how to subscribe to this event
  * @author cpw
+ * @see net.minecraftforge.fml.common.Mod.EventHandler for how to subscribe to this event
  */
-public class FMLPostInitializationEvent extends FMLStateEvent
-{
-    public FMLPostInitializationEvent(Object... data)
-    {
+public class FMLPostInitializationEvent extends FMLStateEvent {
+    public FMLPostInitializationEvent(Object... data) {
         super(data);
     }
 
     @Override
-    public ModState getModState()
-    {
+    public ModState getModState() {
         return ModState.POSTINITIALIZED;
     }
 
     /**
      * Build an object depending on if a specific target mod is loaded or not.
-     *
+     * <p>
      * Usually would be used to access an object from the other mod.
      *
-     * @param modId The modId I conditionally want to build an object for
+     * @param modId     The modId I conditionally want to build an object for
      * @param className The name of the class I wish to instantiate
      * @return An optional containing the object if possible, or null if not
      */
-    public Optional<?> buildSoftDependProxy(String modId, String className, Object... arguments)
-    {
-        if (Loader.isModLoaded(modId))
-        {
+    public Optional<?> buildSoftDependProxy(String modId, String className, Object... arguments) {
+        if (Loader.isModLoaded(modId)) {
             Class<?>[] args = Arrays.stream(arguments).filter(Objects::nonNull).map(Object::getClass).toArray(Class<?>[]::new);
-            try
-            {
-                Class<?> clz = Class.forName(className,true,Loader.instance().getModClassLoader());
+            try {
+                Class<?> clz = Class.forName(className, true, Loader.instance().getModClassLoader());
                 Constructor<?> ct = clz.getConstructor(args);
                 return Optional.of(ct.newInstance(arguments));
-            }
-            catch (Exception e)
-            {
+            } catch (Exception e) {
                 FMLLog.log.info("An error occurred trying to build a soft depend proxy", e);
                 return Optional.empty();
             }
