--- ../src-base/minecraft/net/minecraftforge/fml/common/event/FMLServerAboutToStartEvent.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/event/FMLServerAboutToStartEvent.java
@@ -25,28 +25,27 @@
 /**
  * Called before the server begins loading anything. Called after {@link FMLPostInitializationEvent} on the dedicated
  * server, and after the player has hit "Play Selected World" in the client. Called before {@link FMLServerStartingEvent}.
- *
+ * <p>
  * You can obtain a reference to the server with this event.
- * @see net.minecraftforge.fml.common.Mod.EventHandler for how to subscribe to this event
+ *
  * @author cpw
+ * @see net.minecraftforge.fml.common.Mod.EventHandler for how to subscribe to this event
  */
 public class FMLServerAboutToStartEvent extends FMLStateEvent {
 
     private MinecraftServer server;
 
-    public FMLServerAboutToStartEvent(Object... data)
-    {
+    public FMLServerAboutToStartEvent(Object... data) {
         super(data);
         this.server = (MinecraftServer) data[0];
     }
+
     @Override
-    public ModState getModState()
-    {
+    public ModState getModState() {
         return ModState.AVAILABLE;
     }
 
-    public MinecraftServer getServer()
-    {
+    public MinecraftServer getServer() {
         return server;
     }
 }
