--- ../src-base/minecraft/net/minecraftforge/fml/common/network/simpleimpl/IMessageHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/simpleimpl/IMessageHandler.java
@@ -25,11 +25,10 @@
  * process your packet. Supply the class to {@link SimpleNetworkWrapper#registerMessage(Class, Class, int, net.minecraftforge.fml.relauncher.Side)}
  * to register both the message type and it's associated handler.
  *
- * @author cpw
- *
- * @param <REQ> This is the request type - it is the message you expect to <em>receive</em> from remote.
+ * @param <REQ>   This is the request type - it is the message you expect to <em>receive</em> from remote.
  * @param <REPLY> This is the reply type - it is the message you expect to <em>send</em> in reply. You can use IMessage as the type here
- * if you don't anticipate sending a reply.
+ *                if you don't anticipate sending a reply.
+ * @author cpw
  */
 public interface IMessageHandler<REQ extends IMessage, REPLY extends IMessage> {
     /**
