--- ../src-base/minecraft/net/minecraftforge/event/entity/player/SleepingLocationCheckEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/SleepingLocationCheckEvent.java
@@ -30,26 +30,23 @@
 /**
  * This event is fired when game checks, if sleeping player should be still considered "in bed".<br>
  * Failing this check will cause player to wake up.<br>
- *
+ * <p>
  * This event has a result. {@link HasResult}<br>
- *
+ * <p>
  * setResult(ALLOW) informs game that player is still "in bed"<br>
  * setResult(DEFAULT) causes game to check {@link Block#isBed(IBlockState, IBlockAccess, BlockPos, Entity)} instead
  */
 @HasResult
-public class SleepingLocationCheckEvent extends PlayerEvent
-{
+public class SleepingLocationCheckEvent extends PlayerEvent {
 
     private final BlockPos sleepingLocation;
 
-    public SleepingLocationCheckEvent(EntityPlayer player, BlockPos sleepingLocation)
-    {
+    public SleepingLocationCheckEvent(EntityPlayer player, BlockPos sleepingLocation) {
         super(player);
         this.sleepingLocation = sleepingLocation;
     }
 
-    public BlockPos getSleepingLocation()
-    {
+    public BlockPos getSleepingLocation() {
         return sleepingLocation;
     }
 }
