--- ../src-base/minecraft/net/minecraftforge/server/command/CommandTps.java
+++ ../src-work/minecraft/net/minecraftforge/server/command/CommandTps.java
@@ -26,6 +26,7 @@
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.DimensionType;
+import net.minecraft.world.WorldServer;
 import net.minecraftforge.common.DimensionManager;
 
 class CommandTps extends CommandBase
