--- ../src-base/minecraft/net/minecraftforge/event/LootTableLoadEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/LootTableLoadEvent.java
@@ -30,41 +30,34 @@
  * This event is fired whenever resources are loaded, or when the server starts.
  * This event will NOT be fired for LootTables loaded from the world folder, these are
  * considered configurations files and should not be modified by mods.
- *
+ * <p>
  * Canceling the event will make it load a empty loot table.
- *
  */
 @Cancelable
-public class LootTableLoadEvent extends Event
-{
+public class LootTableLoadEvent extends Event {
     private final ResourceLocation name;
     private LootTable table;
     private LootTableManager lootTableManager;
 
-    public LootTableLoadEvent(ResourceLocation name, LootTable table, LootTableManager lootTableManager)
-    {
+    public LootTableLoadEvent(ResourceLocation name, LootTable table, LootTableManager lootTableManager) {
         this.name = name;
         this.table = table;
         this.lootTableManager = lootTableManager;
     }
 
-    public ResourceLocation getName()
-    {
+    public ResourceLocation getName() {
         return this.name;
     }
 
-    public LootTable getTable()
-    {
+    public LootTable getTable() {
         return this.table;
     }
 
-    public LootTableManager getLootTableManager()
-    {
+    public LootTableManager getLootTableManager() {
         return this.lootTableManager;
     }
 
-    public void setTable(LootTable table)
-    {
+    public void setTable(LootTable table) {
         this.table = table;
     }
 }
