--- ../src-base/minecraft/net/minecraft/client/renderer/EnumFaceDirection.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/EnumFaceDirection.java
@@ -5,8 +5,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
-public enum EnumFaceDirection
-{
+public enum EnumFaceDirection {
     DOWN(new EnumFaceDirection.VertexInformation[]{new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.SOUTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.SOUTH_INDEX)}),
     UP(new EnumFaceDirection.VertexInformation[]{new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.SOUTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.SOUTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.NORTH_INDEX)}),
     NORTH(new EnumFaceDirection.VertexInformation[]{new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.EAST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.DOWN_INDEX, EnumFaceDirection.Constants.NORTH_INDEX), new EnumFaceDirection.VertexInformation(EnumFaceDirection.Constants.WEST_INDEX, EnumFaceDirection.Constants.UP_INDEX, EnumFaceDirection.Constants.NORTH_INDEX)}),
@@ -17,23 +16,19 @@
     private static final EnumFaceDirection[] FACINGS = new EnumFaceDirection[6];
     private final EnumFaceDirection.VertexInformation[] vertexInfos;
 
-    public static EnumFaceDirection getFacing(EnumFacing facing)
-    {
+    public static EnumFaceDirection getFacing(EnumFacing facing) {
         return FACINGS[facing.getIndex()];
     }
 
-    private EnumFaceDirection(EnumFaceDirection.VertexInformation[] vertexInfosIn)
-    {
+    private EnumFaceDirection(EnumFaceDirection.VertexInformation[] vertexInfosIn) {
         this.vertexInfos = vertexInfosIn;
     }
 
-    public EnumFaceDirection.VertexInformation getVertexInformation(int index)
-    {
+    public EnumFaceDirection.VertexInformation getVertexInformation(int index) {
         return this.vertexInfos[index];
     }
 
-    static
-    {
+    static {
         FACINGS[EnumFaceDirection.Constants.DOWN_INDEX] = DOWN;
         FACINGS[EnumFaceDirection.Constants.UP_INDEX] = UP;
         FACINGS[EnumFaceDirection.Constants.NORTH_INDEX] = NORTH;
@@ -43,28 +38,25 @@
     }
 
     @SideOnly(Side.CLIENT)
-    public static final class Constants
-        {
-            public static final int SOUTH_INDEX = EnumFacing.SOUTH.getIndex();
-            public static final int UP_INDEX = EnumFacing.UP.getIndex();
-            public static final int EAST_INDEX = EnumFacing.EAST.getIndex();
-            public static final int NORTH_INDEX = EnumFacing.NORTH.getIndex();
-            public static final int DOWN_INDEX = EnumFacing.DOWN.getIndex();
-            public static final int WEST_INDEX = EnumFacing.WEST.getIndex();
-        }
+    public static final class Constants {
+        public static final int SOUTH_INDEX = EnumFacing.SOUTH.getIndex();
+        public static final int UP_INDEX = EnumFacing.UP.getIndex();
+        public static final int EAST_INDEX = EnumFacing.EAST.getIndex();
+        public static final int NORTH_INDEX = EnumFacing.NORTH.getIndex();
+        public static final int DOWN_INDEX = EnumFacing.DOWN.getIndex();
+        public static final int WEST_INDEX = EnumFacing.WEST.getIndex();
+    }
 
     @SideOnly(Side.CLIENT)
-    public static class VertexInformation
-        {
-            public final int xIndex;
-            public final int yIndex;
-            public final int zIndex;
+    public static class VertexInformation {
+        public final int xIndex;
+        public final int yIndex;
+        public final int zIndex;
 
-            private VertexInformation(int xIndexIn, int yIndexIn, int zIndexIn)
-            {
-                this.xIndex = xIndexIn;
-                this.yIndex = yIndexIn;
-                this.zIndex = zIndexIn;
-            }
+        private VertexInformation(int xIndexIn, int yIndexIn, int zIndexIn) {
+            this.xIndex = xIndexIn;
+            this.yIndex = yIndexIn;
+            this.zIndex = zIndexIn;
         }
+    }
 }
