--- ../src-base/minecraft/net/minecraft/util/Mirror.java
+++ ../src-work/minecraft/net/minecraft/util/Mirror.java
@@ -1,7 +1,6 @@
 package net.minecraft.util;
 
-public enum Mirror
-{
+public enum Mirror {
     NONE("no_mirror"),
     LEFT_RIGHT("mirror_left_right"),
     FRONT_BACK("mirror_front_back");
@@ -9,18 +8,15 @@
     private final String name;
     private static final String[] mirrorNames = new String[values().length];
 
-    private Mirror(String nameIn)
-    {
+    private Mirror(String nameIn) {
         this.name = nameIn;
     }
 
-    public int mirrorRotation(int rotationIn, int rotationCount)
-    {
+    public int mirrorRotation(int rotationIn, int rotationCount) {
         int i = rotationCount / 2;
         int j = rotationIn > i ? rotationIn - rotationCount : rotationIn;
 
-        switch (this)
-        {
+        switch (this) {
             case FRONT_BACK:
                 return (rotationCount - j) % rotationCount;
             case LEFT_RIGHT:
@@ -30,26 +26,19 @@
         }
     }
 
-    public Rotation toRotation(EnumFacing facing)
-    {
+    public Rotation toRotation(EnumFacing facing) {
         EnumFacing.Axis enumfacing$axis = facing.getAxis();
         return (this != LEFT_RIGHT || enumfacing$axis != EnumFacing.Axis.Z) && (this != FRONT_BACK || enumfacing$axis != EnumFacing.Axis.X) ? Rotation.NONE : Rotation.CLOCKWISE_180;
     }
 
-    public EnumFacing mirror(EnumFacing facing)
-    {
-        switch (this)
-        {
+    public EnumFacing mirror(EnumFacing facing) {
+        switch (this) {
             case FRONT_BACK:
 
-                if (facing == EnumFacing.WEST)
-                {
+                if (facing == EnumFacing.WEST) {
                     return EnumFacing.EAST;
-                }
-                else
-                {
-                    if (facing == EnumFacing.EAST)
-                    {
+                } else {
+                    if (facing == EnumFacing.EAST) {
                         return EnumFacing.WEST;
                     }
 
@@ -58,14 +47,10 @@
 
             case LEFT_RIGHT:
 
-                if (facing == EnumFacing.NORTH)
-                {
+                if (facing == EnumFacing.NORTH) {
                     return EnumFacing.SOUTH;
-                }
-                else
-                {
-                    if (facing == EnumFacing.SOUTH)
-                    {
+                } else {
+                    if (facing == EnumFacing.SOUTH) {
                         return EnumFacing.NORTH;
                     }
 
@@ -77,12 +62,10 @@
         }
     }
 
-    static
-    {
+    static {
         int i = 0;
 
-        for (Mirror mirror : values())
-        {
+        for (Mirror mirror : values()) {
             mirrorNames[i++] = mirror.name;
         }
     }
