--- ../src-base/minecraft/net/minecraftforge/client/IRenderContextHandler.java
+++ ../src-work/minecraft/net/minecraftforge/client/IRenderContextHandler.java
@@ -23,14 +23,14 @@
  * @deprecated TODO remove in 1.13. This has never been used
  */
 @Deprecated
-public interface IRenderContextHandler
-{
-    /** Run before the specified rendering context.
+public interface IRenderContextHandler {
+    /**
+     * Run before the specified rendering context.
      */
     void beforeRenderContext();
 
-    /** Run after the specified rendering context.
+    /**
+     * Run after the specified rendering context.
      */
     void afterRenderContext();
 }
-
