--- ../src-base/minecraft/net/minecraft/client/gui/GuiScreen.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiScreen.java
@@ -146,7 +146,7 @@
         {
             if (i == 0)
             {
-                list.set(i, p_191927_1_.getItem().getForgeRarity(p_191927_1_).getColor() + (String)list.get(i));
+                list.set(i, p_191927_1_.getRarity().rarityColor + (String)list.get(i));
             }
             else
             {
